## **One - Hot Encodding Method**

In [ ]:
import pandas as pd

# Load the dataset
df = pd.read_csv('/content/cleaned_dataset.csv')

# Display the first few rows to verify
print(df.head())


   Unnamed: 0  count  hate_speech  offensive_language  neither  class  \
0           0      3            0                   0        3      2   
1           1      3            0                   3        0      1   
2           2      3            0                   3        0      1   
3           3      3            0                   2        1      1   
4           4      6            0                   6        0      1   

                                               tweet  \
0  retweet as a woman you should not complain abo...   
1  retweet boy dats coldtyga down bad for cuffin ...   
2  retweet dawg retweet you ever fuck a bitch and...   
3                     retweet she look like a tranny   
4  retweet the shit you hear about me might be tr...   

                                        tweet_tokens  
0  ['retweet', 'woman', 'complain', 'cleaning', '...  
1  ['retweet', 'boy', 'dat', 'coldtyga', 'bad', '...  
2  ['retweet', 'dawg', 'retweet', 'ever', 'fuck',...  
3   

In [ ]:
import ast

# Convert string representation of list to actual list
df['tweet_tokens'] = df['tweet_tokens'].apply(ast.literal_eval)
# Display the first few rows to verify
print("\nDataFrame after converting tweet_tokens to lists:")

print(df.head())



DataFrame after converting tweet_tokens to lists:
   Unnamed: 0  count  hate_speech  offensive_language  neither  class  \
0           0      3            0                   0        3      2   
1           1      3            0                   3        0      1   
2           2      3            0                   3        0      1   
3           3      3            0                   2        1      1   
4           4      6            0                   6        0      1   

                                               tweet  \
0  retweet as a woman you should not complain abo...   
1  retweet boy dats coldtyga down bad for cuffin ...   
2  retweet dawg retweet you ever fuck a bitch and...   
3                     retweet she look like a tranny   
4  retweet the shit you hear about me might be tr...   

                                        tweet_tokens  
0  [retweet, woman, complain, cleaning, house, an...  
1  [retweet, boy, dat, coldtyga, bad, cuffin, dat...  
2  [retw

In [ ]:
# Define feature and target variables
X = df[['tweet_tokens']]
y = df['class']



from sklearn.feature_extraction.text import CountVectorizer
from sklearn.compose import ColumnTransformer

# Define a function to join tokens into a single string
X['tweet_tokens'] = X['tweet_tokens'].apply(lambda x: ' '.join(x))

# Use CountVectorizer for the 'tweet_tokens' column
column_transformer = ColumnTransformer(
    transformers=[
        ('tweet_tokens', CountVectorizer(), 'tweet_tokens')  # Note the use of CountVectorizer
    ],
    remainder='passthrough'  # This keeps the other columns as is
)

<ipython-input-3-76cd7a50a8c6>:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  X['tweet_tokens'] = X['tweet_tokens'].apply(lambda x: ' '.join(x))


In [ ]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.pipeline import make_pipeline

# Create a Random Forest Classifier pipeline
pipeline = make_pipeline(column_transformer, RandomForestClassifier())



from sklearn.model_selection import train_test_split

# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Display the training data
print("\nTrain Data:")
print(X_train.head())
print(y_train.head())

# Display the testing data
print("\nTest Data:")
print(X_test.head())
print(y_test.head())



Train Data:
                                            tweet_tokens
15272  retweet well else white people get u forget ho...
9351   funny thing isits people people seeing pic jud...
20323  retweet nigga messed wrong bitch andand andand...
3638                                    bitch a nigggaaa
20579                                         real bitch
15272    0
9351     2
20323    1
3638     1
20579    1
Name: class, dtype: int64

Test Data:
                                            tweet_tokens
2281                            got missed call yo bitch
15914  retweet fucking bad bitch gone need money lil ...
18943  retweet laughing loud credit near good know ri...
16407  retweet wipe cum forgot retweet contact lens w...
13326  nigga cheat bitch expect pay back whatsoever n...
2281     1
15914    1
18943    2
16407    1
13326    1
Name: class, dtype: int64


In [ ]:
# Training the model
pipeline.fit(X_train, y_train)
accuracy = pipeline.score(X_test, y_test)
print(f'Accuracy: {accuracy}')


Accuracy: 0.8840024208190438


Using **One - Hot** Encoding Method we get the accuracy of **88%** for our model.